src/extensions/ExtensionManager.as

```diff
@@ -168,8 +168,14 @@ public class ExtensionManager {
 			for(var b:Object in ext.waiting) {
 				if(ext.waiting[b] == id) {
 					delete ext.waiting[b];
-					(b as Block).response = retval;
-					(b as Block).requestState = 2;
+					b = b as Block;
+					if(b) {  // TLF: we probably don't need this...?
```

Comment:
I would skip the `if` here. If it turns out that there's a non-`Block` object in the `waiting` array, then that indicates a fairly serious bug elsewhere in the code. I'd rather be informed through an exception rather than hiding the bug silently.